modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/dataRenderers.ts

```diff
@@ -40,7 +58,7 @@ export interface DataRenderer {
 }
 
 export interface InternalDataRenderer extends DataRenderer {
-	Component: React.ComponentClass<any>;
+	Component: FC<any>;
```

Comment:
Using FC here might be too strict, what if they pass a Class instead? I think we might have a more "open" type for anything "react-y" which I think would be the type `React.ReactNode`.

I may be wrong on this, it seems to change every new React release

Reply:
What about `ComponentType`? It seems to be what we want:

```ts
type React.ComponentType<P = {}> = React.ComponentClass<P, any> | React.FunctionComponent<P>
```

Reply:
That also may work... I would just make sure to test it in your editor against various ways someone might pass in a component.

TS + React starts to get confusing when you are adding external APIs like this, it's been awhile since I have thought through the best way to add types for these. I might need a refreshed in some TS course for react libraries